sql/2019/10_SEO/10_13.sql

```diff
@@ -0,0 +1,47 @@
+#standardSQL
+
+# custom mobile site (e.g. is mobile diffent response compared to desktop?)
+# check if we get a redirect, and if so to where
+
+#standardSQL
+
+SELECT
```

Comment:
@rviscomi This query is 5.2TB to test. I tried with the sample data (query below) but it seems pages_desktop_1k contains different urls then pages_mobile_1k.

What I try to do is look at the mobile request response URL and the desktop request response URL, and flag if they are different (e.g. a custom mobile site).

Reply:
That's right, a website may only exist in one of the tables. The full mobile table actually has 1M more websites than desktop.

Reply:
I am stuck in this one. To see if mobile is different from mobile, I look at the response and the 'redirectURL' field. If they don't match up, mobile is served a different URL.

Reply:
One possible approach may be to compare the page weights of desktop vs mobile sites having the same URL:

```sql
#standardSQL
SELECT
  APPROX_QUANTILES(desktop_bytes - mobile_bytes, 10) AS bytes_diff
FROM
  (SELECT url, bytesTotal as desktop_bytes FROM `httparchive.summary_pages.2019_07_01_desktop`)
JOIN
  (SELECT url, bytesTotal as mobile_bytes FROM `httparchive.summary_pages.2019_07_01_mobile`)
USING (url)
```

Ideally we'll see fewer bytes on mobile.

Another approach could be to detect media query usage:

```sql
#standardSQL
SELECT
  client,
  num_urls,
  pct_urls
FROM
  `httparchive.blink_features.usage`
WHERE
  yyyymmdd = '20190701' AND
  feature = 'CSSAtRuleMedia'
```

Reply:
The metric is more about answering whether websites serve a 'custom mobile' website. Page weight or media queries don't answer that imho.